src/logic/equiv/list.lean

```diff
@@ -119,26 +120,30 @@ instance fin_pi (n) (π : fin n → Type*) [∀ i, encodable (π i)] : encodable
 of_equiv _ (equiv.pi_equiv_subtype_sigma (fin n) π)
 
 /-- If `α` is encodable, then so is `array n α`. -/
-instance array [encodable α] {n} : encodable (array n α) :=
+instance _root_.array.encodable [encodable α] {n} : encodable (array n α) :=
 of_equiv _ (equiv.array_equiv_fin _ _)
 
 /-- If `α` is encodable, then so is `finset α`. -/
-instance finset [encodable α] : encodable (finset α) :=
+instance _root_.finset.encodable [encodable α] : encodable (finset α) :=
 by haveI := decidable_eq_of_encodable α; exact
  of_equiv {s : multiset α // s.nodup}
   ⟨λ ⟨a, b⟩, ⟨a, b⟩, λ ⟨a, b⟩, ⟨a, b⟩, λ ⟨a, b⟩, rfl, λ ⟨a, b⟩, rfl⟩
 
+/-- When `α` is finite and `β` is encodable, `α → β` is encodable too. Because the encoding is not
+unique, we wrap it in `trunc` to preserve computability. -/
 def fintype_arrow (α : Type*) (β : Type*) [decidable_eq α] [fintype α] [encodable β] :
```

Comment:
Could consider
```suggestion
def _root_.function.trunc_encodable_of_fintype (α : Type*) (β : Type*) [decidable_eq α] [fintype α] [encodable β] :
```
but it's not a clear improvement

Reply:
Given that I will (try to) generalize it soon, let's not bother here.

Reply:
Can we add a todo for this so it doesn't get forgotten about?